src/test/sstruct.c

```diff
@@ -3249,6 +3250,24 @@ main( hypre_int argc,
                }
             }
          }
+#else    // Use AddFEMBoxValues
+         /* TODO: There is probably a smarter way to do this copy */
+         for (i = 0; i < data.max_boxsize; i++)
+         {
+            j = i * data.fem_nsparse;
+            hypre_TMemcpy(&d_values[j], data.fem_values, HYPRE_Real,
```

Comment:
@rfalgout this memcpy is taking most of the assembly time. I have some ideas to refactor this, later we can discuss. Thanks!

Reply:
This copy is just for the simplicity of our test driver. The real apps should have their fem data on device already. Should we be concerned by the cost?

Reply:
I agree the apps will already have their data on the device. However, I think we should give "a good example" here if users want to see how to employ the new AddFEMBoxValues routines. Moreover, I think we should consider passing in data relative to a FEM stencil in a contiguous fashion other than interleaved as we are doing currently. This would eliminate strided copies.

Reply:
I think this can wait for a later PR, but let's discuss at one of our upcoming meetings.  Thanks for taking a close look at the timing breakdown!